### ISSUE TITLE
Debug builds fail with arch's new makepkg.conf CFLAGS

### ISSUE THREAD

**Issue by skwerlman** [Cockatrice/Cockatrice#4382]
**System Information:**
Arch Linux
pacman: 6.0.0-2
gcc: 11.1.0-1

_______________________________________________________________________________________

Starting with pacman 6, arch's makepkg.conf builds packages with the following flags:
```
CFLAGS="-march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions \
        -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security \
        -fstack-clash-protection -fcf-protection"
```
https://github.com/archlinux/svntogit-packages/blob/packages/pacman/trunk/makepkg.conf#L41-L43

Cockatrice debug builds negate the `-O2` flag by setting `-O0`. This causes debug builds to fail, since `-Wp,-D_FORTIFY_SOURCE=2` warns when optimization is disabled and `-Werror` turns that into an error.

```
/usr/include/features.h:397:4: error: #warning _FORTIFY_SOURCE requires compiling with optimization (-O) [-Werror=cpp]
  397 | #  warning _FORTIFY_SOURCE requires compiling with optimization (-O)
      |    ^~~~~~~
```

This issue affects debug-mode PKGBUILDs _ONLY_, not builds on arch in general.

_______________________________________________________________________________________

**Steps to reproduce:**
 - update to pacman 6
 - replace makepkg.conf with makepkg.conf.pacnew
 - `cmake -DCMAKE_BUILD_TYPE=Debug ..`
 - `make`

**Comment by tooomm** [comment]
Is this still the case and the issue valid?

**Comment by skwerlman** [comment]
yes, cockatrice still overrides `-O2` but not `-Wp,-D_FORTIFY_SOURCE=2` in debug builds, so if the latter is set the build will fail

**Comment by ebbit1q** [comment]
should we check for the existing -O flag or should we override -Wp -D_FORTIFY_SOURCE=2

**Comment by skwerlman** [comment]
if forcing a specific `-O` level is desired for debug builds, then flags which conflict with that should probably also be overridden.

at least from the arch packaging perspective, overriding `-O` is incorrect<sup>[1](https://wiki.archlinux.org/title/CMake_package_guidelines#CMake_can_automatically_override_the_default_compiler_optimization_flag)</sup>, but i can see the value in overriding it and dont really have a strong opinion either way.
